todolist/server/src/main/scala/handlers/TagRpcServiceHandler.scala

```diff
@@ -17,49 +17,64 @@
 package examples.todolist.server
 package handlers
 
-import cats.Monad
-import cats.syntax.functor._
```

Comment:
The deletion of this import (and similar ones in other files) is causing a lot of your compile errors.

for-comprehensions are syntactic sugar for a chain of `flatMap` and `map`s. e.g.

```scala
    for {
      _     <- L.debug(s"Trying to reset $model in repository")
      items <- repo.init
      _     <- L.warn(s"Reset $model table in repository")
    } yield items.map(MessageId)
```

desugars to

```scala
L.debug(s"Trying to reset $model in repository").flatMap { _ =>
  repo.init.flatMap { items =>
    L.warn(s"Reset $model table in repository").map { _ =>
      items.map(MessageId)
    }
  }
}
```

Importing `cats.syntax.functor._` will add `map` as an extension method on any type that has a `Functor` instance. Similarly `import cats.syntax.flatMap._` will add a `flatMap` extension method to any type that has a `FlatMap` instance.

In this case we have `F[_]: Sync`, so we know that any type `F[something]` will have both `Functor` and `FlatMap` instances. So these two imports should be enough to get the for-comprehensions to compile.

Reply:
Oh my gosh of course!!  Thank you for catching that; you're exactly right; once I added those extension methods most of the errors disappeared.  

That said, now I'm left with an interesting error that I imagine is of the same variety and I'd love your take on it:

I have the following for-comprehension:
```
override def insert(item: TodoListRequest): F[TodoListResponse] =
    for {
      _            <- L.debug(s"Trying to insert a $model")
      insertedItem <- repo.insert(item.toTodoList)
      _            <- L.info(s"Tried to add a $model")
      todoListItem <- insertedItem.map(_.toTodoList)
    } yield todoListItem
```

which seems to follow in terms of types (insertedItem is of type `F[Option[TodoList]]`, and then `toTodoList` is a helper function that looks like this:
```
implicit class OptionTodoListTodoListResponse(ol: Option[TodoList]) {

  def toTodoList: TodoListResponse =
    TodoListResponse(ol.flatMap(_.toTodoListMessage))
```

so it _should_ be able to map the `insertedItem` to a `TodoListResponse`.

However, what I get instead is a compiler error that says
```
type mismatch;
 found   : Option[Nothing]
 required: examples.todolist.protocol.Protocols.TodoListResponse
```

Is this because technically `insertedItems` isn't of type `Option[TodoList]` but instead `[F[Option[[TodoList]]`, and I need some sort of addition extension method from cats to correctly handle that type signature?  That's my guess, but I could be missing something else.

Reply:
ARGH never mind I cracked it; I just need to call `insertedItem.toTodoList` rather than using `map`.